bosh-director/lib/bosh/director/deployment_plan/instance_plan.rb

```diff
@@ -80,6 +81,9 @@ def needs_recreate?
           if @recreate_deployment
             @logger.debug("#{__method__} job deployment is configured with \"recreate\" state")
             true
+          elsif @need_to_fix
```

Comment:
I think it would be better for InstancePlan to check the state of the existing instance directly, similar to how it does all the other checks on the raw models/objects - this way we keep the logic centralized in this class and based on state. Otherwise we start relying on external, InstancePlan instantiators to perform the correct logic and correctly influence how instances are potentially going to be changing later. Maybe others have some thoughts though...